chemprop/data/data.py

```diff
@@ -221,6 +221,19 @@ def bond_types(self) -> List[List[float]]:
         :return: A list of bond types for each molecule.
         """
         return [[b.GetBondTypeAsDouble() for b in self.mol[i].GetBonds()] for i in range(self.number_of_molecules)]
+    @property
+    def molecular_weight(self) -> float:
+        """
+        Gets the maximum molecular weight among all the molecules in the :class:`MoleculeDatapoint`.
+
+        :return: The maximum molecular weight.
+        """
+        max_molecular_weight = 0.0  # Initialize to 0 or any suitable starting value
+        for mol in self.mol:
+            molecular_weight = Chem.rdMolDescriptors.CalcExactMolWt(mol)
+            if molecular_weight > max_molecular_weight:
+                max_molecular_weight = molecular_weight
+        return max_molecular_weight
```

Comment:
```suggestion
        return max(Chem.rdMolDescriptors.CalcExactMolWt(mol) for mol in self.mol)
```
n.b. not sure if it's possible for any in `self.mol` to be `None`

Reply:
done. thanks for the suggestion.